source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRTextBlockRenderer.mm

```diff
@@ -195,6 +196,41 @@ + (NSNumber *)getTextStrokeWidthForWeight:(TextWeight)weight
             return @-2;
     }
 }
+// given string with short form of date (MM/dd/yyyy), convert it to
+// NSDateFormatterMediumStyle or NSDateFormatterLongStyle in local language
++ (void) getLocalizedDate:(std::shared_ptr<TextBlock> const &)txtBlck stringWithDate:(std::string &)dateParsedString
+{
+    std::vector<std::shared_ptr<DateTimePreparsedToken>> DateTimePreparsedTokens =  DateTimePreparser(txtBlck->GetText()).GetTextTokens();
+    for(auto section : DateTimePreparsedTokens){
+        if(section->GetFormat() != DateTimePreparsedTokenFormat::RegularString) {
+            NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+            [formatter setDateFormat:@"MM/dd/yyy"];
+            std::string givenDate = std::to_string(section->GetMonth()) + "/" +  std::to_string(section->GetDay()) + "/" +  std::to_string(section->GetYear());
+            NSString *nsString = [NSString stringWithCString:givenDate.c_str() encoding:NSUTF8StringEncoding];
+            NSDate *date = [formatter dateFromString:nsString];
+            // specify output date format
+            NSDateFormatter *outputFormatter = [[NSDateFormatter alloc] init];
+
+            outputFormatter.timeStyle = NSDateFormatterNoStyle;
+            if(section->GetFormat() == DateTimePreparsedTokenFormat::DateShort){
+                outputFormatter.dateStyle = NSDateFormatterShortStyle;
+            } else if(section->GetFormat() == DateTimePreparsedTokenFormat::DateCompact){
```

Comment:
Short date should be formatted with NSDateFormatterMediumStyle as the output should be similar to "Mon, Feb 13th, 2017"

Reply:
thanks

---
In reply to: [171447206](https://github.com/Microsoft/AdaptiveCards/pull/1194#discussion_r171447206) [](ancestors = 171447206)